client/src/components/brushableHistogram/histogram.tsx

```diff
@@ -130,7 +132,31 @@ const Histogram = ({
                 i: number
               ) => string
             )
-        );
+        )
+        .selectAll("text")
+        .each((_, i, d3Nodes) => {
+          const d3Node = d3.select(d3Nodes[i]);
+          const text = d3Node.text();
+          const node = d3Node.node() as SVGTextElement;
+          let textLength = node.getComputedTextLength?.();
+          let truncatedText = text;
+
+          /**
+           * (thuang): Truncate the text if it's too long to fit in the histogram.
+           */
+          if (textLength > TRUNCATION_MAX_WIDTH_PX) {
```

Comment:
it will decrease the text length one char at a time until it fits into the max width. The runtime is horrible, but we're operating on 6 to 8 strings on average, so probably okay.

If optimization is needed, maybe binary could help!

Reply:
I wonder if there is something sophisticated that could be done with css `text-overflow: ellipsis;` https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/. This seems fine for now though!

Reply:
Ah dang, okay. it looks like svg text element cannot be styled with width 😭 

https://stackoverflow.com/questions/15975440/add-ellipses-to-overflowing-text-in-svg

So we'll need to use the JS solution still!